File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        while i < len(nums):
            if i == 0 or nums[i] > nums[i - 1]:
                i += 1
            else:
                nums.pop(i)
        return len(nums)
            

File: Data Structures & Algorithms/monotonic-array/submission-0.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        /*
            keep a diff of 0.
            if diff is negative and current diff of nums[i] and nums[j]
            is positive OR diff is positive and current diff is negative,
            return false.
            return true if loop finishes.
        */
        int diff = 0; 
        for (int j = 1; j < nums.length; j++) {
            int i = j - 1;
            int currDiff = nums[j] - nums[i];
            if (currDiff > 0 && diff < 0) {
                return false;
            }
            if (currDiff < 0 && diff > 0) {
                return false;
            }
            diff += currDiff;
        }
        return true;
    }
}